libs/estdlib/src/sets.erl

```diff
@@ -0,0 +1,453 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 1996-2024. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+%
+% This is shrinked down version of OTP sets module, that supports just the new format
+%
+
+-module(sets).
+-compile({no_auto_import, [size/1]}).
+
+-export_type([set/0, set/1]).
+
+-export([
+    new/0, new/1,
+    is_set/1,
+    size/1,
+    is_empty/1,
+    from_list/1,
+    from_list/2,
+    is_equal/2,
+    to_list/1,
+    is_element/2,
+    add_element/2,
+    del_element/2,
+    union/1,
+    union/2,
+    intersection/1,
+    intersection/2,
+    is_disjoint/2,
+    subtract/2,
+    is_subset/2,
+    fold/3,
+    filter/2,
+    map/2
+]).
+
+-define(VALUE, []).
+
+-type set() :: set(_).
+
+%-doc "As returned by `new/0`.".
```

Comment:
I’m not sure how well this will render with edoc. The edoc code style markdown tags should all be changed from:
```
`example/0`
```
to:
```
`example/0'
```

I don’t know what system was used to generate the documentation for this style of code. I have a feeling this current documentation will not render well.

Reply:
They were commented out because I need to fix them